packages/context/src/resolution-session.ts

```diff
@@ -258,15 +299,40 @@ export class ResolutionSession {
    * Get the binding path as `bindingA --> bindingB --> bindingC`.
    */
   getBindingPath() {
-    return this.bindings.map(b => b.key).join(' --> ');
+    return this.stack
+      .filter(i => i.type === 'binding')
+      .map(b => (<Binding>b.value).key)
+      .join(' --> ');
   }
 
   /**
-   * Get the injection path as `injectionA->injectionB->injectionC`.
+   * Get the injection path as `injectionA --> injectionB --> injectionC`.
    */
   getInjectionPath() {
-    return this.injections
-      .map(i => ResolutionSession.describeInjection(i)!.targetName)
+    return this.stack
+      .filter(i => i.type === 'injection')
+      .map(
+        i =>
+          ResolutionSession.describeInjection(<Injection>i.value)!.targetName,
```

Comment:
Is this type cast still required?

Reply:
Yes. TS is not smart enough to understand the array filtering criteria.